### PR TITLE
Replace Spire references with SPIFFE Workload Api

### LINKED ISSUE #2: Does this work with SPIRE, or any service that exposes the SPIFFE Workload API?
This project is described as the SPIFFE Envoy Agent, and only appears to require the SPIFFE Workload API in order to be used. But the `README.md` refers repeatedly to SPIRE. Is SPIRE strictly necessary, or would this work with any vendor that implements the SPIFFE Workload API.

If the latter, the documentation should avoid mentioning SPIRE specifically. If the former, maybe the project should be called `spire-envoy-agent`?

### PR DESCRIPTION
**Pull request by MarcosDY**
Signed-off-by: Marcos Yacob <marcos@scytale.io>

**Description of change**
Update readme to no longer metion SPIRE Agent, and replace it with SPIFFE Workload api.

**Which issue this PR fixes**
Fixes #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ This component allows a user to integrate Envoy proxy with the SPIFFE identity f
 ## How does this integration work?
 
 This agent implements two services:
-- [Secret discovery service (SDS)](https://www.envoyproxy.io/docs/envoy/latest/configuration/secret#config-secret-discovery-service): Used to provide Envoy with x509-SVIDs fetched from SPIRE.
+- [Secret discovery service (SDS)](https://www.envoyproxy.io/docs/envoy/latest/configuration/secret#config-secret-discovery-service): Used to provide Envoy with x509-SVIDs fetched from SPIFFE Workload API client.
```
Comment (azdagron) [inline README.md]: ...fetched from the SPIFFE Workload API.

File: README.md [hunk 2]
```diff
@@ -16,21 +16,21 @@ This agent implements two services:
 
 ## Communication between components
 
-When spiffe-envoy-agent starts, it registers itself with the above components: SPIRE agent and Envoy proxy. Agent communication is done using a [SPIFFE Workload API client](https://github.com/spiffe/spire/tree/master/api/workload) over a unix domain socket (or UDS). On the other hand, Envoy communication is ruled by Envoy [xDS protocol](https://github.com/envoyproxy/data-plane-api/blob/master/XDS_PROTOCOL.md) which also uses a UDS to communicate with spiffe-envoy-agent.
+When spiffe-envoy-agent starts, it registers itself with the above components: SPIFFE Workload API and Envoy proxy. Agent communication is done using a [SPIFFE Workload API client](https://github.com/spiffe/spire/tree/master/api/workload) over a unix domain socket (or UDS). On the other hand, Envoy communication is ruled by Envoy [xDS protocol](https://github.com/envoyproxy/data-plane-api/blob/master/XDS_PROTOCOL.md) which also uses a UDS to communicate with spiffe-envoy-agent.
```
Comment (azdagron) [inline README.md]: Hmm, spiffe-envoy-agent doesn't register itself with Envoy proxy. I'd suggest.

"When spiffe-envoy-agent starts, it registers with the SPIFFE Workload API and starts listening for requests from Envoy. Communication with the SPIFFE Workload API is done using gRPC over a unix domain socket (or UDS). Envoy uses the xDS API over UDS to talk to the spiffe-envoy-agent."

File: README.md [hunk 3]
```diff
@@ -16,21 +16,21 @@ This agent implements two services:
...
 
 ### Transport layer: X509-SVID
 
- For [x509-SVID](https://github.com/spiffe/spiffe/blob/master/standards/X509-SVID.md) management, the spiffe-envoy-agent exposes a [secret discovery service](https://www.envoyproxy.io/docs/envoy/latest/configuration/secret#config-secret-discovery-service), or SDS. A stream is created between it and SPIRE agent using the SPIFFE Workload API. SPIRE updates the stream with new SVIDs according to the configured settings.
+ For [x509-SVID](https://github.com/spiffe/spiffe/blob/master/standards/X509-SVID.md) management, the spiffe-envoy-agent exposes a [secret discovery service](https://www.envoyproxy.io/docs/envoy/latest/configuration/secret#config-secret-discovery-service), or SDS. A stream is created between it and SPIFFE Workload API client using the SPIFFE Workload API. SPIFFE Workload API updates the stream with new SVIDs according to the configured settings.
```
Comment (azdagron) [inline README.md]: "A stream is created between the spiffe-envoy-agent and the SPIFFE Workload API. The SPIFFE Workload API updates the stream...."

File: README.md [hunk 4]
```diff
@@ -16,21 +16,21 @@ This agent implements two services:
...
- At the same time, Envoy initiates the SDS communication with a [discovery request](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/discovery.proto#discoveryrequest) message. The spiffe-envoy-agent establishes a stream and sends Envoy the latest certificate and key in a [discovery response](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/discovery.proto#discoveryresponse) message. Certificate updates are then communicated in successive responses every time SPIRE rotates the SVID. Once Envoy applies the configurations received, it is ready to handle TLS connections.
+ At the same time, Envoy initiates the SDS communication with a [discovery request](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/discovery.proto#discoveryrequest) message. The spiffe-envoy-agent establishes a stream and sends Envoy the latest certificate and key in a [discovery response](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/discovery.proto#discoveryresponse) message. Certificate updates are then communicated in successive responses every time SPIFFE Workload API rotates the SVID. Once Envoy applies the configurations received, it is ready to handle TLS connections.
```
Comment (azdagron) [inline README.md]: "the SPIFFE Workload API"?

**Comment by ajessup** [comment]
LGTM aside from @azdagron's comments

File: README.md [hunk 5]
```diff
@@ -16,13 +16,13 @@ This agent implements two services:
 
 ## Communication between components
 
-When spiffe-envoy-agent starts, it registers itself with the above components: SPIFFE Workload API and Envoy proxy. Agent communication is done using a [SPIFFE Workload API client](https://github.com/spiffe/spire/tree/master/api/workload) over a unix domain socket (or UDS). On the other hand, Envoy communication is ruled by Envoy [xDS protocol](https://github.com/envoyproxy/data-plane-api/blob/master/XDS_PROTOCOL.md) which also uses a UDS to communicate with spiffe-envoy-agent.
+When spiffe-envoy-agent starts, it registers with the SPIFFE Workload API and starts listening for requests from Envoy. Communication with the [SPIFFE Workload API](https://github.com/spiffe/spire/tree/master/api/workload) is done using gRPC over a unix domain socket (or UDS). Envoy uses the [xDS API](https://github.com/envoyproxy/data-plane-api/blob/master/XDS_PROTOCOL.md) over UDS to talk to the spiffe-envoy-agent.
```
Comment (azdagron) [inline README.md]: the SPIFFE Workload API link is to the client library, maybe change it to be to the workload API proto?

Comment (MarcosDY) [reply README.md]: totally miss that one!!! I'll updarte it to `https://github.com/spiffe/spire/blob/master/proto/api/workload/workload.proto`

**Pull request merged**